core/pkg/sync-store/sync_store.go

```diff
@@ -132,13 +131,16 @@ func (s *SyncStore) RegisterSubscription(
 			errChan:  errChan,
 			dataSync: dataSync,
 		}
-
 		// access pointer + trigger resync passing the dataSync
 		if sh.syncRef != nil {
 			go func() {
-				s.logger.Debug(fmt.Sprintf("sync handler exists for target %s, triggering a resync", target))
-				if err := sh.syncRef.ReSync(ctx, dataSync); err != nil {
-					errChan <- err
+				s.mu.Lock()
```

Comment:
can we use `RLock/RUnlock` here? It seems like the syncHandlers are not modified heare, so a read lock might be sufficient

Reply:
Ya, agreed with @bacherfl , this seems like only read-ops.